src/renderers/webgl/WebGLTextures.js

```diff
@@ -1594,6 +1594,15 @@ function WebGLTextures( _gl, extensions, state, properties, capabilities, utils,
 		const renderTargetProperties = properties.get( renderTarget );
 		const isCube = ( renderTarget.isWebGLCubeRenderTarget === true );
 
+		// check if the depth texture is already bound to the frame buffer and that it's been initialized
+		if ( renderTargetProperties.__boundDepthTexture === renderTarget.depthTexture && properties.has( renderTarget.depthTexture ) ) {
```

Comment:
Can we move this check in `setRenderTarget()` and do it before checking the dimensions of the depth texture? That check is only relevant when exchanging depth textures and not in the default case.

Reply:
I realized this is a bit more complicated when I was thinking through it again. Hopefully the current change makes sense. We need account for the case the where a depth texture is used and already bound on a render target, then disposed - meaning the depth texture is implicitly unbound so we need to make sure the three.js state accounts for this. I tested and pushed the current change to the depth peeling demo page.